File: python/짝지어제거하기.py
def solution(s):
    
    if len(s) % 2 == 1:
        return 0
    
    stack = []
    for i in range(len(s)):
        if len(stack) < 1:
            stack.append(s[i])
        elif stack[-1] == s[i]:
            stack.pop()
        else:
            stack.append(s[i])
    
    if len(stack) > 0:
        return 0
    return 1

print(solution('baabaa'))
print(solution('cdcd'))

File: JavaScript/짝지어제거하기.js
function solution(s) {
    const basket = [];
    for (let i = 0; i < s.length; i++) {
        basket.push(s[i]);
        if (basket.length >= 2 && basket[basket.length - 1] === basket[basket.length - 2]) {
            basket.splice(basket.length - 2, 2);
        }
    }
    return basket.length === 0 ? 1 : 0;
}
